lib/transport/abstract-transport.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+var util = require("util");
+var EventEmitter = require('events');
+
+var AbstractTransport = function () {
+	if (this.constructor === AbstractTransport) {
+		throwBecauseAbstract();
```

Comment:
I'm not sure there's a point in having an AbstractTransport class at all, if it's not used to enforce implementation outside of the amqp10 library. I understand the ugliness of the circular dependency, but it has its usefulness. if we're duck typing outside implementations, why not do it for ours too. after all, the built-in implementations are the safest